tests/TestCase/Database/Schema/SqliteSchemaTest.php

```diff
@@ -261,6 +261,47 @@ protected function _createTables($connection): void
         $connection->execute('CREATE INDEX "created_idx" ON "schema_articles" ("created")');
         $connection->execute('CREATE UNIQUE INDEX "unique_id_idx" ON "schema_articles" ("unique_id")');
 
+        $table = <<<SQL
+CREATE TABLE schema_no_rowid_pk (
+id INT PRIMARY KEY
+);
+SQL;
+        $connection->execute($table);
+
+        $table = <<<SQL
+CREATE TABLE schema_unique_constraint_variations (
+id INTEGER PRIMARY KEY AUTOINCREMENT,
+no_quotes INTEGER,
+'single_''quotes' INTEGER,
+"double_""quotes" INTEGER,
+`tick_``quotes` INTEGER,
+foo INTEGER,
+bar INTEGER,
+baz INTEGER,
+CONSTRAINT no_quotes_idx UNIQUE (no_quotes)
+CONSTRAINT duplicate_idx UNIQUE (no_quotes)
+CONSTRAINT 'single_''quotes_idx' UNIQUE ('single_''quotes')
+CONSTRAINT "double_""quotes_idx" UNIQUE ("double_""quotes")
+CONSTRAINT `tick_``quotes_idx` UNIQUE (`tick_``quotes`)
+CONSTraint
+    a_cat_walked_over_my_keyboard_idx     UNIque
+        (    id  ,'foo',
+            	    "bar",     `baz`)
+);
+SQL;
+        $connection->execute($table);
+
+        $table = <<<SQL
+CREATE TABLE schema_foreign_key_variations (
+id INTEGER PRIMARY KEY AUTOINCREMENT,
+author_id INT(11),
+author_name VARCHAR(50),
+CONSTRAINT author_fk FOREIGN KEY (author_id) REFERENCES schema_authors (id) ON UPDATE CASCADE ON DELETE RESTRICT
```

Comment:
Yes and no, sorry, I forgot to mention it, there is some incompatibility introduced here. I've tested it manually, and it would detect changes, it would generate drops for the "old" indexes/constraints, and added the "new" ones.

Not great, not terrible, that shouldn't be overly disruptive. The problem with Sqlite here specifically is however that dropping foreign keys [is currently disabled](https://github.com/cakephp/migrations/blob/3.x/src/Table.php#L156-L174), which creates quite some problems in and of itself. People using Sqlite always had manually intervene in these cases. I had planned on FK dropping ever since I've worked on Phinx's Sqlite adapter, and, well... I guess I should test this again, and possibly add test cases for it to check what exactly happens, and then we can asses whether we want to change this or not.

Migrations is, well, something... introducing new behavior or fixes always has a high chance of introducing BC incompatibilities. Question is at what point we want to pull of the plasters and put some disinfectant on the wounds.

Reply:
Oh and for unique indexes/constraints specifically, they don't work currently anyways, whether it's diffing or snapshotting, the `sqlite_autoindex_` names would always cause SQL errors, meaning just like with dropping foreign keys, people using Sqlite would have had to manually intervene.

I don't wanna say "_they're used to the pain anyways"_, but I kinda want to say it 😛

Reply:
Interesting, I didn't know that sqlite support for migrations was in that rough of shape. This will be a step in the right direction then. At least we're not reflecting indexes that can't be re-created anymore.

Reply:
> [...] I had planned on enabling FK dropping ever since I've worked on Phinx's Sqlite adapter, and, well... I guess I should test this again, and possibly add test cases for it to check what exactly happens here, and then we can asses whether we want to change this or not.

Looking at it again I now remember that it involved more than just enabling it, as Migrations currently generates drops by foreign key names instead of columns in `up()` (as opposed to drops by columns in `down()` 🙃), and Phinx's Sqlite adapter doesn't support dropping FKs by name currently.

I'd need to enable diff baking for Sqlite too then, which I'd wanted to spare for the endgame, as I expect it to uncover new problems. @markstory If you'd like to see that first, then I'd give it a try, otherwise I'd do it afterwards in case this PR gets accepted.

Reply:
No we can do it afterwards. I think this improves our sqlite support in useful ways, so if we cause a bit of short term chaos it will be worth it the long term.